src/app/select-list/select-list.component.ts

```diff
@@ -14,15 +14,27 @@ export class SelectListComponent implements OnInit {
     options: selectListItem[];
 
     title: string;
+    multiSelect?: boolean;
+    onSelect?: (selectedItem: selectListItem, selected?: boolean, list?: selectListItem[]) => void,
   }
   ngOnInit() {
   }
   selected: selectListItem;
   select(item: selectListItem) {
-    this.selected = item;
-    this.close();
+    if (!this.args.multiSelect) {
+      this.selected = item;
+      if (this.args.onSelect) 
+        this.args.onSelect(item, true);
+      this.close();
+    }
   }
 
+  onChange(item: selectListItem) {
```

Comment:
this mixes concerns between the multi select dialog and the select helpers screen - there is a mixup here between the select etc....